src/components/Filters/index.tsx

```diff
@@ -0,0 +1,245 @@
+			onFiltersChange?.(newFilters);
+		},
+		[filters, onFiltersChange],
+	);
+
+	const handleSaveView = React.useCallback(
+		(name: string) => {
+			const view: FiltersView = {
+				id: randomUUID(),
+				name,
+				eventName: `Saving filters view ${name}`,
+				filters: filters ?? [],
+			};
+			const newViews = [...(views ?? []), view];
+			onViewsChange?.(newViews);
+		},
+		[filters, views, onViewsChange],
+	);
+
+	const handleDeleteView = React.useCallback(
+		(view: FiltersView) => {
+			const newViews = views?.filter((v) => v.id !== view.id);
+			onViewsChange?.(newViews ?? []);
+		},
+		[views, onViewsChange],
+	);
+
+	return (
+		<>
+			<Box display="flex" alignItems="center" gap={2}>
+				{(!renderMode || renderMode.includes('search')) && (
+					<Box
+						ref={searchBarRef}
+						sx={{
+							display: 'flex',
+							flex: 1,
+							position: 'relative',
+							alignSelf: 'center',
+							zIndex: 10,
+						}}
+					>
+						<Search
+							id="balena__filters_search_bar"
+							focused={dark}
+							InputProps={{
+								style: { color: dark ? 'white' : 'primary' },
+							}}
+							onChange={(e) => {
+								if (!e) {
+									setSearchString('');
+									setShowSearchDropDown(false);
+									return;
+								}
+								setShowSearchDropDown(e.target.value !== '');
+								setSearchString(e.target.value);
+							}}
+							onEnter={(event) => {
+								event.preventDefault();
+								event.stopPropagation();
+								if (!searchString) {
+									return;
+								}
+								const filter = createFullTextSearchFilter(schema, searchString);
+								handleFilterChange(filter);
+								setSearchString('');
+							}}
+							value={searchString}
+						/>
+						{searchString && showSearchDropDown && onSearch?.(searchString)}
+					</Box>
+				)}
+				{(!renderMode || renderMode.includes('add')) && (
+					<Button
+						variant={dark ? 'light' : 'outlined'}
+						color="secondary"
+						onClick={() => setShowFilterDialog(true)}
+						startIcon={matches ? <FontAwesomeIcon icon={faFilter} /> : null}
+						sx={{ whiteSpace: 'nowrap' }}
+					>
+						{matches ? (
+							t('actions.add_filter')
+						) : (
+							<FontAwesomeIcon icon={faFilter} />
+						)}
+					</Button>
+				)}
+				{(!renderMode || renderMode.includes('views')) && (
+					<Views
+						views={views}
+						setFilters={(filters) => onFiltersChange?.(filters)}
+						deleteView={handleDeleteView}
+						dark={dark}
+					/>
+				)}
+			</Box>
+			{(!renderMode || renderMode.includes('summary')) && filters?.length ? (
+				<Summary
+					onEdit={(filter) => {
+						setEditFilter(filter);
+					}}
+					onDelete={handleDeleteFilter}
+					onClearFilters={() => {
+						onFiltersChange?.([]);
+					}}
+					showSaveView={showSaveView}
+					onSaveView={({ name }) => handleSaveView(name)}
+					filters={filters}
+					dark={dark}
+				/>
+			) : null}
+			{(showFilterDialog || formData) && (
+				<FiltersDialog
+					open={showFilterDialog}
```

Comment:
We don't need this since we rely on `(showFilterDialog || formData)` to render the dialog. `open` should just be manually set to true in the `FiltersDialog`.

Furthermore, the way it is right now, if `showFilterDialog` is falsey and `formData` is truthy, the dialog *still* won't open since `open` at the moment only depends on `showFilterDialog`

Reply:
we can set it to true in the dialog, you right.